src/main/java/SortingHat.java

```diff
@@ -0,0 +1,15 @@
+public class SortingHat {
+    public static ItemUpdater assign(Item item) {
+        if (AgedBrie.NAME.equals(item.getName())) {
```

Comment:
Consider refactoring to AgedBrie.matches(item) for added semantic and flexibility. Also consider to put the matches() inside an enum instead of statics of a class. Extra brownie points if you can link the actual item classes with the enum through reflection.